### DISCUSSION TITLE
pg adapter not working?

### DISCUSSION THREAD

**Discussion by wing-it** [oguimbal/pg-mem#154] [Q&A] [answered]
The `pg` adapter doesn't appear to be working correctly for me, I feel like I may be missing some obvious step.

Initializing the adapter seems to work as intended:
<code>
import {newDb} from 'pg-mem';
const {Client} = newDb().adapters.createPg();
</code>

However, whenever I try to do anything with `Client` such as `connect()` or `query('...')` I get `TypeError: client.query is not a function`. Am I doing something wrong or are these basic methods not supported with this adapter?

**Comment by wing-it** [comment, marked as answer]
Yep, this was 100% me missing a detail, when you get this adapter back you still need to initialize it with `new Client();`. My bad.

**Reply by fdelcaztws24** [reply]
What do you mean? I dont understand exactly and I have the same problem

**Reply by Strajk** [reply]
@wing-it thx, i've struggled with the same issue and this discussion helped 🙏 

my working code
```
const db = newDb()
const pg = db.adapters.createPg()
const pgClient = new pg.Client() // the new is important
```

if you are using the code from [Wiki](https://github.com/oguimbal/pg-mem/wiki/Libraries-adapters)
```
import {newDb} from 'pg-mem';
const {Client} = newDb().adapters.createPg();
```
u have u instantiate it
```
const client = new Client()
```
